proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/sdk/PowerAuthSDK.java

```diff
@@ -1638,4 +1641,70 @@ public void onFingerprintInfoDialogClosed() {
+     *     <li>You have to always call {@code cancel()} on provided {@link ICancelable} object once the operation is finished,
+     *         otherwise the thread will be blocked indefinitely.
+     *     <li>It doesn't matter whether the actual HTTP request is performed directly in {@code onExecution()} method, or you'll use
+     *         your own thread. What's matter is to call {@code cancel()} once the operation is done.
+     * </ul>
+     *
+     * Check the documentation for {@link IAsyncOperation} for more details.
+     *
+     * @param asynchronousOperation {@link IAsyncOperation} to be executed on a serialized thread executor.
+     * @return {@link ICancelable} object, allowing the external cancellation.
+     * @throws PowerAuthErrorException if there's not a valid activation.
+     */
+    public @NonNull ICancelable executeOnSerialExecutor(@NonNull final IAsyncOperation asynchronousOperation) throws PowerAuthErrorException {
+        final CancelableAsynchronousOperation operation = new CancelableAsynchronousOperation(asynchronousOperation);
+        executeOnSerialExecutor(operation);
+        return operation;
+    }
+
+    /**
+     * Executes provided {@link Runnable} object on the internal, serialized thread executor. This gives application an opportunity to serialize
+     * its own signed HTTP requests, with requests created in the SDK internally.
+     *
+     * <h3>Why this matters</h3>
+     *
+     * The PowerAuth SDK is using that executor for serialization of signed HTTP requests, to guarantee, that only one request is processed
+     * at the time. The PowerAuth signatures are based on a logical counter, so this technique makes that all requests are delivered
+     * to the server in the right order. So, if the application is creating its own signed requests, then it's recommended to synchronize
+     * them with the SDK.
+     *
+     * <h3>Recommended practices</h3>
+     * <ul>
+     *     <li>You should calculate PowerAuth signature from the {@link Runnable#run()} method.
+     *     <li>{@link Runnable#run()} should return from its execution after the HTTP request is fully processed, or at least after
+     *         the response headers are received (e.g. you know that server already did process the request)
+     * </ul>
+     *
+     * Unlike the {@link #executeOnSerialExecutor(IAsyncOperation)} method, this variant is more generic. You can, for example, execute
+     * your own networking {@link AsyncTask} directly, but be aware that the PowerAuth signature should be calculated in {@link AsyncTask#doInBackground(Object[])}
+     * method.
+     *
+     * @param runnable {@link Runnable} operation to be executed on a serialized thread executor.
+     * @throws PowerAuthErrorException if there's not a valid activation.
+     */
+    public void executeOnSerialExecutor(@NonNull final Runnable runnable) throws PowerAuthErrorException {
```

Comment:
I guess Runnable is enough here.
Or we can expose the executor and let them schedule the async tasks there.